calendar/calendar-tool/tool/src/webapp/vm/calendar/chef_calendar_view.vm

```diff
@@ -0,0 +1,130 @@
+#set ($dollar = "$")
+<div class="portletBody">
+  #if($menu)#toolbar($menu)#end
+  <div class="page-header">
+    <h1>$tlang.getString("tool_title")</h1>
+    <div class="sakai-sideHeading">
+      <a href="$printableVersionUrl" title="$!tlang.getString('java.print')" target="_blank" rel="noreferrer">$!tlang.getString('java.print')</a>
+    </div>
+  </div>
+  #if ($alertMessage)<div class="sak-banner-warn">$tlang.getString('gen.alert') $formattedText.escapeHtml($alertMessage)</div>#end
+  <div class="sakai-table-toolBar">
+    <div class="sakai-table-filterContainer">
+      #calendarView()
+    </div>
+  </div>
+  ## Calendar DIV where the full calendar will be rendered
+  <div id="calendarDiv"></div>
+  ## include the Legend:
+  #calendarLegend()
+</div>
+
+<link href="/library/webjars/fullcalendar/5.8.0/main.min.css" rel="stylesheet" />
+<script src="/library/webjars/fullcalendar/5.8.0/main.min.js"></script>
+<script src="/library/webjars/fullcalendar/5.8.0/locales-all.min.js"></script>
+<script>
+  var userLanguage = portal.locale || navigator.language;
+  // Get the event color from the skin variables.
+  var computedStyle = getComputedStyle(document.querySelector(':root'));
+  var eventBackgroundColor = computedStyle.getPropertyValue('--infoBanner-bgcolor', '#e9f5fc');
+  var eventTextColor = computedStyle.getPropertyValue('--infoBanner-color', '#196390');
+  var sakaiOrigin = window.location.origin;
+  var siteId = portal.siteId;
+  // We need the updated calendar events, do not cache requests.
+  var requestHeaders = new Headers();
+  requestHeaders.append('pragma', 'no-cache');
+  requestHeaders.append('cache-control', 'no-cache');
+  var requestInit = {
+    method: 'GET',
+    headers: requestHeaders,
+  };
+  var calendarDiv = document.getElementById('calendarDiv');
+  // Initialize fullcalendar and render it in the calendarDiv.
+  var calendar = new FullCalendar.Calendar(calendarDiv, {
+    initialView: 'timeGridWeek',
+    displayEventTime: false,
+    headerToolbar: {
+      left: 'prev,next today',
+      center: 'title',
+      right: 'dayGridMonth,timeGridWeek,timeGridDay,listWeek'
+    },
+    eventSources: [{
+      events: function(event, successCallback, failureCallback) {
+        var start_date = moment(event.start).format('YYYY-MM-DD');
+        var end_date = moment(event.end).format('YYYY-MM-DD');
+        var requestData = {
+          merged: true,
+          firstDate: start_date,
+          lastDate: end_date
+        };
+        // Fetch the site calendar events from the REST endpoint.
+        var calendarRequestUrl = new URL(`${sakaiOrigin}/direct/calendar/site/${siteId}.json`);
+        Object.keys(requestData).forEach(key => calendarRequestUrl.searchParams.append(key, requestData[key]));
+        window.fetch(calendarRequestUrl, requestInit)
+        .then(response => response.json())
```

Comment:
You need to handle network errors in here.
r => {
  if (r.ok) return r.json();
  else throw new Error("Network error while fetching calendar data");
})
...
.catch(error => console.error(error));

or something like that..

Reply:
Yeah that makes sense.